lib/matplotlib/ticker.py

```diff
@@ -2621,8 +2621,14 @@ def __call__(self):
             return []
 
         if self.ndivs is None:
-            x = int(np.round(10 ** (np.log10(majorstep) % 1)))
-            if x in [1, 5, 10]:
+
+            majorstep_no_exponent = 10 ** (np.log10(majorstep) % 1)
+
+            # The part after the 'or' is for #13069 to make minimal changes to
+            # behaviour
+            if np.isclose(majorstep_no_exponent, 2.5) or (
+                    (not np.isclose(majorstep_no_exponent, [1, 10]).any() and
+                     int(np.round(majorstep_no_exponent)) in [1, 5, 10])):
```

Comment:
The two conditions (not close to 1 or 10, but possibly in [1 or 10]) seem contradictory.
Given that you are already going to use round(), I would just use its ability to round to one decimal point:
```
majorstep_no_exponent = round(10**(np.log10(majorstep) % 1), 1)
# (in fact the following looks more natural to me, but heh)
majorstep_no_exponent = round(majorstep / 10**int(np.log(majorstep)), 1)
if majorstep_no_exponent in [2.5, 5]: ...
else: ...
```

Reply:
Thanks @anntzer for the comments.
1) The conditions are not close to 1 or 10 and when **rounded** are in [1 or 10], so not always contradictory (for example 1.1 satisfies both). 
I was trying to make as minimal a change to the number of minor ticks calculation. 
Rounding to 1 decimal place:
```
if round(majorstep_no_exponent, 1) in [2.5, 5]:
    ndivs = 5
else:
    ndivs = 4
```
will change the number of minor divisions from 5 to 4 for major ticks spaced 1.1 units apart (for example), compared to the situation before this PR. @jklymak indicated this is undesirable. The code is a bit ugly because of this, though.
2) I think the calculation used ```%``` in order to guarantee 1 <= majorstep_no_exponent <= 10 because that's the range of the steps in AutoLocator. ```majorstep / 10**int(np.log10(majorstep))``` can return values outside this range (for example if marjostep is 0.5).

Reply:
Re 1: I personally think if we're already changing the minorstep mapping for the common case of 1/5/10-spaced ticks (which appear by default), it's really not a big leap to also change the case of 1.1-spaced ticks (which don't even appear by default).  This just needs to be documented... (plus, I'm pretty sure there's a lot of other places in the code which probably don't work so well with "weird" tick spacings like 1.1)
Re 2: Sorry, that should have been `m/10**np.floor(np.log10(m))` (int rounds towards zero, but I want to round down), which works for m<1 too.  Anyways, the formulas are equivalent, no big deal.

Reply:
Simplified condition according to @anntzer's suggestion.